.circleci/config.yml

```diff
@@ -16,4 +20,31 @@ jobs:
             - ./node_modules
       - run: npm run lint
       - run: npm test
-      # See https://circleci.com/docs/2.0/deployment-integrations/ for deploy examples
+      - persist_to_workspace:
+          root: ~/repo
+          paths: .
+
+  deploy:
+    <<: *common
+    steps:
+      - attach_workspace:
+          at: ~/repo
+      - run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/repo/.npmrc
+      - run: npm publish
+
+workflows:
+  version: 2
+  build-deploy:
+    jobs:
+      - build:   # run for all branches and PRs
+          filters:
+            tags:
+              only: /.*/
+      - deploy: # run for only tagged (master)
+          requires:
+            - build
+          filters:
+            tags:
+              only: /^v.*/
```

Comment:
Will this cause a problem if for some reason we have a tag name that starts with a `v` but isn't a version number? should we make this regex more specific?

Reply:
In general I've only used git tags for versions in the master branch of repos, and that must be pretty common because all the CircleCI examples are assuming it too. You may have noticed the seemingly useless `/.*/` match everything in the `build filters tags` step above, it's needed to make this work right. Maybe you're getting the impression that I'm not a CircleCI fan, and you'd be right 😼, but all these CI systems seem to have crazy quirks.